### PR TITLE
Add email domain validation for organization discovery

### PR DESCRIPTION
**Pull request by Rashmini**
### Purpose
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/17297

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 1]
```diff
@@ -261,6 +297,11 @@ const AddOrganizationDiscoveryDomains: FunctionComponent<AddOrganizationDiscover
                                             } }
                                             { ...params }
                                             margin="dense"
+                                            error={ isEmailDomainDataError }
+                                            helperText= { 
+                                                isEmailDomainDataError
+                                            && emailDomainDataError
```
Comment (Achintha444) [inline apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:303]:
```suggestion
                                                isEmailDomainDataError
                                                  ? emailDomainDataError
                                                  : null
```

Comment (Achintha444) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx:303]: Check other places

**Comment by Achintha444** [comment]
Add `core` and `myaccount` to the changeset as well.
https://github.com/wso2/identity-apps/blob/master/docs/release/README.md

File: apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx [hunk 2]
```diff
@@ -169,6 +172,25 @@ const EditOrganizationDiscoveryDomains: FunctionComponent<EditOrganizationDiscov
...
+     *
+     * @param values - Email domains.
+     */
+    const validateEmailDomain = (emailDomainList: string[]) => {
+
+        const emailDomainValidation: boolean = FormValidation.domain(emailDomainList[emailDomainList.length-1]);
+
+        if (!emailDomainValidation) {
+            setIsEmailDomainDataError(true);
+            setEmailDomainDataError( t(
```
Comment (pavinduLakshan) [inline apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx:186]: Seems like this is the only value being set to emailDomainDataError. WDYT of removing the state variable for error msg and directly using the error message in other places.

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx (hunk 1 above)
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx]: Fixed with 59ca7063d0007513c62b470cde9b726f017412ee

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 3]
```diff
@@ -147,14 +156,41 @@ const AddOrganizationDiscoveryDomains: FunctionComponent<AddOrganizationDiscover
...
+                        )
+                    })
+                );
             });
     };
 
+    /**
+     * Function to validate the input string is an email domain.
+     *
+     * @param values - Email domains.
```
[the remark this answers is not in the document]
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx]: Fixed with 59ca7063d0007513c62b470cde9b726f017412ee

File: apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx [hunk 4]
```diff
@@ -147,14 +156,41 @@ const AddOrganizationDiscoveryDomains: FunctionComponent<AddOrganizationDiscover
...
+     */
+    const validateEmailDomain = (emailDomainList: string[]) => {
+
+        const emailDomainValidation: boolean = FormValidation.domain(emailDomainList[emailDomainList.length-1]);
```
[the remark this answers is not in the document]
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx]: Fixed with 59ca7063d0007513c62b470cde9b726f017412ee

File: apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx (hunk 2 above)
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx]: Fixed with 59ca7063d0007513c62b470cde9b726f017412ee

File: apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx [hunk 5]
```diff
@@ -169,6 +172,25 @@ const EditOrganizationDiscoveryDomains: FunctionComponent<EditOrganizationDiscov
         return <ContentLoader dimmer />;
     }
 
+    /**
+     * Function to validate the input string is an email domain.
+     *
+     * @param values - Email domains.
+     */
+    const validateEmailDomain = (emailDomainList: string[]) => {
```
[the remark this answers is not in the document]
Comment (Rashmini) [reply apps/console/src/features/organization-discovery/components/edit-organization-discovery-domains.tsx:179]: This function is not reusable. The reusable method is added to FormValidation.

**Comment by Rashmini** [comment]
> Add `core` and `myaccount` to the changeset as well. https://github.com/wso2/identity-apps/blob/master/docs/release/README.md

Added

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**